lib/types.go

```diff
@@ -41,6 +42,18 @@ type Options struct {
 	Thresholds map[string][]*Threshold `json:"thresholds"`
 }
 
+type SourceData struct {
```

Comment:
Or actually, the SourceData struct needs to include a type to work. getSrcData() auto-detects the type and has to return it in some way, along with the source data and the filename. I guess I can stop using the constants TypeJS and TypeURL and create new constants instead (e.g. SourceData_JS, SourceData_URL), if the issue is mainly that this functionality should not re-use constant types meant to identify a certain type of runner?